### ISSUE TITLE
Error Upgrading to Frescobaldi 3.1.3

### ISSUE THREAD

**Issue by ggivler** [frescobaldi/frescobaldi#1368]
I downloaded and installed version 3.1.3 of Frescobaldi, when I run this on an older .ly file I get the following error, this is on a Mac:

Here is the system information:

Hardware Overview:

  Model Name:	Mac mini
  Model Identifier:	Macmini8,1
  Processor Name:	Intel Core i7
  Processor Speed:	3.2 GHz
  Number of Processors:	1
  Total Number of Cores:	6
  L2 Cache (per Core):	256 KB
  L3 Cache:	12 MB
  Hyper-Threading Technology:	Enabled
  Memory:	32 GB

Here is the stacktrace.

Traceback (most recent call last):
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 141, in engraveRunner
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 145, in engravePreview
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 200, in engrave
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 252, in runJob
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/manager.py", line 61, in start_job
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/__init__.py", line 211, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/lilypond.py", line 176, in configure_command
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 138, in wait
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 261, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 278, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 189, in set
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 191, in emit
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 308, in call
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 276, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 306, in frommacports
TypeError: unsupported operand type(s) for +: 'bool' and 'str'

Let me know if you more information.

**Comment by mattlibera** [comment]
Not sure if this is the same situation, but I was seeing this after installing Frescobaldi on a new machine, where I had not yet installed Lilypond. I installed Lilypond and this went away.

**Comment by Stevearzh** [comment]
I met the same problem, I install Frescobaldi by homebrew, and I already installed Lilypond. But I still met the problem when I tried to engrave .ly file. Here is the error information:

```
Frescobaldi: 3.1.3
Extension API: 0.9.0
Python: 3.9.1
python-ly: 0.9.7
Qt: 5.11.3
PyQt: 5.15.2
sip: 4.19.24
poppler: 20.12.1
python-poppler-qt: 0.75.0
OS: macOS-10.16-x86_64-i386-64bit
installation kind: standalone .app bundle

Traceback (most recent call last):
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 141, in engraveRunner
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 145, in engravePreview
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 200, in engrave
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 252, in runJob
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/manager.py", line 61, in start_job
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/__init__.py", line 211, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/lilypond.py", line 176, in configure_command
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 138, in wait
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 261, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 278, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 189, in set
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 191, in emit
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 308, in call
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 276, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 306, in frommacports
TypeError: unsupported operand type(s) for +: 'bool' and 'str'



```

**Comment by fpetrogalli** [comment]
I am not sure why, but he issue seems to be linked to the fact that for some reasons Frescobaldi isn't able to find the `lilypond` executable. Specifying the full path of the `lilypond` executable in the preferences of Frescobaldi made the trick for me (the error has disappeared).

**Comment by tsgouros** [comment]
I think I see the problem, which is that Lilypondinfo.bindir() in lilypondinfo.py is set either to return the location of the bin directory or False. It seems to be designed only to be used with an "if self.bindir()..." before it, within the class. There is no such if statement in the 'Lilypondinfo.frommacports" method, so it returns False and the function tries to append False to '/ports'.

I can't supply a tested patch because I don't really understand how to test my fixes. I haven't dived into the how-to-build-on-mac yet. But I think that changing frommacports() to read like this should help:

```
         if self.bindir():
                portbin = os.path.abspath(self.bindir() + '/port')
            else:
                return False
```

Also, my version of Frescobaldi (and Lilypond) was installed via Homebrew, not MacPorts. Not sure how that should be reflected in this file. The path for lilypond on my machine is /opt/homebrew/bin...

**Comment by dliessi** [comment]
This issue is another duplicate of #1386, which was solved by #1397 in both `master` and `v3.1.x`.
The next version of Frescobaldi, either 3.1.4 or 3.2, will include the fix.
In the meantime you can live with your change, or manually apply the changes of #1397; see https://github.com/frescobaldi/frescobaldi/issues/1386#issuecomment-1004786920 for further information.
